feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go

```diff
@@ -173,6 +175,51 @@ func TestInterfaceCounters(t *testing.T) {
 	}
 }
 
+func validateInAndOutPktsPerSecond(t *testing.T, dut *ondatra.DUTDevice, i1, i2 *interfaces.InterfacePath) bool {
+	if deviations.InterfaceCountersFromContainer(dut) {
+		time.Sleep(10 * time.Second)
+		return true
+	}
+	inSamples := gnmi.Collect(t, dut.GNMIOpts().WithYGNMIOpts(ygnmi.WithSubscriptionMode(gpb.SubscriptionMode_SAMPLE), ygnmi.WithSampleInterval(30*time.Second)), i1.Counters().InUnicastPkts().State(), 90*time.Second)
+	outSamples := gnmi.Collect(t, dut.GNMIOpts().WithYGNMIOpts(ygnmi.WithSubscriptionMode(gpb.SubscriptionMode_SAMPLE), ygnmi.WithSampleInterval(30*time.Second)), i2.Counters().OutUnicastPkts().State(), 90*time.Second)
+
+	inPkts := inSamples.Await(t)
+	outPkts := outSamples.Await(t)
+
+	if len(inPkts) < 2 || len(outPkts) < 2 {
+		t.Fatalf("did not get enough samples: in counters: %s out counters %s",
+			inPkts, outPkts)
+	}
+	t.Logf("Sample Size Incoming Packets: %d, Sample Size Outgoing Packets: %d", len(inPkts), len(outPkts))
+	var pktCounterOK = true
+	// check counters for first and last sample interval, they shouldn't be equal
+	inValFirst, _ := inPkts[0].Val()
+	outValFirst, _ := outPkts[0].Val()
+	inValFinal, _ := inPkts[len(inPkts)-1].Val()
+	outValFinal, _ := outPkts[len(inPkts)-1].Val()
+
+	if inValFinal == inValFirst || outValFinal == outValFirst {
+		t.Logf("Counters not incremented: Initial Incoming Packets: %d, Final Incoming Packets: %d", inValFirst, inValFinal)
+		t.Logf("Counters not incremented: Initial Outgoing Packets: %d,  Final Outgoing Packets: %d", outValFirst, outValFinal)
+		pktCounterOK = false
+		return pktCounterOK
+	}
+
+	for i := 1; i < len(inPkts); i++ {
+		inValOld, _ := inPkts[i-1].Val()
+		outValOld, _ := outPkts[i-1].Val()
+		inValLatest, _ := inPkts[i].Val()
+		outValLatest, _ := outPkts[i].Val()
+		t.Logf("Incoming Packets: %d, Outgoing Packets: %d", inValLatest, outValLatest)
+		if inValLatest-inValOld != outValLatest-outValOld {
```

Comment:
I think we also want to check that each value pair has increased.   Bug we want to detect is if the old and latest are the same value.

Reply:
for example, if inValOld is 100 and inValLatest is 100, and outValLatest and outValOld is also 100,  yet we are sending traffic, then the test should fail.

Reply:
Done